src/samplers/lumigoSampler.ts

```diff
@@ -13,68 +12,109 @@ export class LumigoSampler implements Sampler {
+    return endpoint_attr ? endpoint_attr.toString() : null;
+  } else if (spanKind === SpanKind.SERVER) {
+    const endpoint_attr = attributes['url.path'] || attributes['http.target'];
+    return endpoint_attr ? endpoint_attr.toString() : null;
   }
 
-  // Try building a raw url from given attributes
-  let raw_url = null;
-  if (attributes['http.url']) {
-    raw_url = attributes['http.url'].toString();
-  }
-  if (
-    !raw_url &&
-    attributes['http.scheme'] &&
-    attributes['http.host'] &&
-    attributes['http.target']
-  ) {
-    raw_url = `${attributes['http.scheme']}://${attributes['http.host']}${attributes['http.target']}`;
+  return null;
+};
+
+export const doesMatchClientSpanFilteringRegexes = (endpoint: string): boolean => {
+  if (!endpoint || !process.env.LUMIGO_FILTER_HTTP_ENDPOINTS_REGEX_CLIENT) {
+    return false;
   }
 
-  if (!raw_url) {
-    return null;
+  const regexes = parseStringToArray(process.env.LUMIGO_FILTER_HTTP_ENDPOINTS_REGEX_CLIENT);
+  return doesEndpointMatchRegexes(endpoint, regexes);
+};
+
+export const doesMatchServerSpanFilteringRegexes = (endpoint: string): boolean => {
+  if (!endpoint || !process.env.LUMIGO_FILTER_HTTP_ENDPOINTS_REGEX_SERVER) {
+    return false;
   }
 
-  return standardizeHttpUrl(raw_url);
+  const regexes = parseStringToArray(process.env.LUMIGO_FILTER_HTTP_ENDPOINTS_REGEX_SERVER);
+  return doesEndpointMatchRegexes(endpoint, regexes);
 };
 
-export const shouldSkipSpanOnRouteMatch = (url: string): boolean => {
-  if (!url) {
+export const doesMatchGeneralSpanFilteringRegexes = (endpoint: string): boolean => {
+  if (!endpoint || !process.env.LUMIGO_FILTER_HTTP_ENDPOINTS_REGEX) {
     return false;
   }
 
-  if (!process.env.LUMIGO_AUTO_FILTER_HTTP_ENDPOINTS_REGEX) {
+  const regexes = parseStringToArray(process.env.LUMIGO_FILTER_HTTP_ENDPOINTS_REGEX);
+  return doesEndpointMatchRegexes(endpoint, regexes);
+};
+
+export const doesEndpointMatchRegexes = (endpoint: string, regexes: string[]): boolean => {
+  if (!endpoint || !regexes) {
     return false;
   }
 
-  if (process.env.LUMIGO_AUTO_FILTER_HTTP_ENDPOINTS_REGEX) {
-    let regex: null | RegExp = null;
+  for (const rawRegex of regexes) {
     try {
-      regex = new RegExp(process.env.LUMIGO_AUTO_FILTER_HTTP_ENDPOINTS_REGEX);
-      return regex.test(url);
+      if (new RegExp(rawRegex).test(endpoint)) {
+        return true;
+      }
     } catch (err) {
-      console.error(
-        `Invalid regex in 'LUMIGO_AUTO_FILTER_HTTP_ENDPOINTS_REGEX': '${process.env.LUMIGO_AUTO_FILTER_HTTP_ENDPOINTS_REGEX}'`
-      );
-      return false;
+      console.error(`Invalid regex: '${rawRegex}', skipping it.`);
     }
   }
+
   return false;
 };
 
+export const parseStringToArray = (rawArray: string): string[] => {
+  try {
+    const parsedArray = JSON.parse(rawArray);
+    if (Array.isArray(parsedArray) && !parsedArray.some((e) => typeof e !== 'string')) {
```

Comment:
I would change it to avoid double negation:
```suggestion
    if (Array.isArray(parsedArray) && !parsedArray.every((e) => typeof e === 'string')) {
```